### ISSUE TITLE
Issue: `maxDepth` parameter not working properly with `Json` fields

### ISSUE THREAD

**Issue by StephanDecker** [maoosi/prisma-appsync#71]
When defining the following prisma schema (see below) and storing nested objects in this JSON it raises the maxDepth Count (which is not correct). Result is ```Query has depth of 11, which exceeds max depth of ....'``` although the nesting takes place in the JSON column and not in Graphql. Could you reproduce this problem?

Event Log from Cloudwatch (path param):
```
  identity: {
    requestApiKey: ....
    requestUserAgent:..
  },
  paths: [
    '/update/targetProfile/modifiedBy',
    '/update/targetProfile/rule/and[0]/and[0]/==[0]/var',
    '/update/targetProfile/rule/and[0]/and[0]/==[1]',
    '/update/targetProfile/rule/and[0]/and[1]/and[0]/some[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[0]/some[1]/and[0]/some[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[0]/some[1]/and[0]/some[1]/==[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[0]/some[1]/and[0]/some[1]/==[1]',
    '/update/targetProfile/rule/and[0]/and[1]/and[0]/some[1]/and[1]/some[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[0]/some[1]/and[1]/some[1]/<=[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[0]/some[1]/and[1]/some[1]/<=[1]',
    '/update/targetProfile/rule/and[0]/and[1]/and[0]/some[1]/and[2]/some[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[0]/some[1]/and[2]/some[1]/>=[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[0]/some[1]/and[2]/some[1]/>=[1]',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[0]/filter[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[0]/filter[1]/and[0]/==[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[0]/filter[1]/and[0]/==[1]',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[0]/filter[1]/and[1]/==[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[0]/filter[1]/and[1]/==[1]',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[0]/filter[1]/and[2]/==[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[0]/filter[1]/and[2]/==[1]',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[0]/filter[1]/and[3]/==[0]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[0]/filter[1]/and[3]/==[1]',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[1]/+[0]',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[1]/+[1]/var',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[0]/reduce[2]',
    '/update/targetProfile/rule/and[0]/and[1]/and[1]/some[1]/>=[1]',
.....
````
https://github.com/maoosi/prisma-appsync/blob/main/packages/client/src/guard.ts#L177-L190

```
model TargetProfile {
    id                     String          @id @default(uuid())
    rule                   Json
}
```

**Comment by maoosi** [comment]
Thanks for reporting the issue @StephanDecker, will add to the roadmap.

**Comment by maoosi** [comment]
Fix will be released as part of `1.0.0-rc.5`.
